### DISCUSSION TITLE
opencv-python

### DISCUSSION THREAD

**Discussion by leabanea** [beeware/mobile-forge#51] [Q&A]
I noticed that the opencv-Issue has been open for a while. Does this mean that it's not currently possible to forge an opencv-python or opencv-python-headless wheel with this tool?

Im a uni student, so I don't have too much experience with this, but from what Ive gathered in my first attempt there's an issue with  cmakes configuration?
Id be really grateful for any tips since  I desperately need cv2 as Ultralytics-import for my thesis project.

**Comment by freakboy3742** [comment]
The fact that there is an open ticket means that, with the current state of the code in this project, it is not possible to build an iOS-compatible wheels using this project.

It should be possible to compile opencv-python manually, or to develop a set of patches for this project to perform such a build, but I can't point you at any particular set of instructions for this.

It's been a while since I looked at this in particular, but my recollection is that the holdup isn't actually opencv itself, but the dependencies of opencv. I believe SciPy is a requirement for building openCV; building SciPy for iOS is problematic because building SciPy requires a Fortran compiler. It *may* be possible to build enough of SciPy to satisfy OpenCV's needs - but again, it's been a while since I've looked at this, and I don't have any ready-to-use instructions I can point you at.

**Reply by mhsmith** [reply]
Going by [the Chaquopy recipe](https://github.com/chaquo/chaquopy/blob/master/server/pypi/packages/opencv-python/meta.yaml), OpenCV doesn't require SciPy, but it does require CMake. That's relatively easy on Android because the NDK comes with a toolchain file that does most of the configuration for us, but I don't know what the best approach would be on iOS.

The Chaquopy recipe also mentions libfortran and OpenBLAS, but I think that was because a previous version of the linker required all the transitively linked libraries to be available during the build, not because OpenCV contains Fortran code itself.

**Comment by sabyasachi1889** [comment]


**Reply by freakboy3742** [reply]
No - they are getting the error because OpenCV - no matter how you spell the package name - doesn't have any packaged wheels for iOS.
